meta/state.go

```diff
@@ -62,6 +64,16 @@ func (r *localRaft) remove() error {
 	return nil
 }
 
+func (r *localRaft) shutdown() error {
```

Comment:
Hmm.  `close()` was intended to shutdown the metastore cleanly when the server shuts down or if you transition from a local raft to a remote raft (e.g. you're demoted but not dropped).  Granted, the latter can't happen yet but that was the intent.

I'm was mainly referring to here though: https://github.com/influxdb/influxdb/blob/master/meta/state.go#L214-L220.  Seems like those lines could instead call `shutdown()` and then set it to `nil` instead of duplicating the logic.